store/sqlstore/thread_store.go

```diff
@@ -501,14 +501,17 @@ func (s *SqlThreadStore) DeleteMembershipForUser(userId string, postId string) e
 	return nil
 }
 
-func (s *SqlThreadStore) CreateMembershipIfNeeded(userId, postId string, following, incrementMentions, updateFollowing bool) error {
+func (s *SqlThreadStore) MaintainMembership(userId, postId string, following, incrementMentions, updateFollowing, updateViewedTimestamp bool) error {
```

Comment:
Can we add a test please? :)

Reply:
test cases already exist in store/storetest/thread_store.go . I've just renamed the function to be clearer and added a param to update a timestamp

Reply:
@agnivade added